src/som/interpreter/Interpreter.java

```diff
@@ -383,44 +370,24 @@ private void send(final SSymbol selector, final SClass receiverClass,
 
     if (invokable != null) {
       // Invoke the invokable in the current frame
-      invokable.invoke(getFrame(), this);
+      invokable.invoke(frame, this);
     } else {
       int numberOfArguments = selector.getNumberOfSignatureArguments();
 
       // Compute the receiver
       SAbstractObject receiver = frame.getStackElement(numberOfArguments - 1);
 
-      receiver.sendDoesNotUnderstand(selector, universe, this);
+      receiver.sendDoesNotUnderstand(selector, universe, this, frame);
     }
   }
 
-  private Frame popFrame() {
-    // Save a reference to the top frame
-    Frame result = frame;
-
-    // Pop the top frame from the frame stack
-    frame = frame.getPreviousFrame();
-
-    // Destroy the previous pointer on the old top frame
-    result.clearPreviousFrame();
-
-    // Return the popped frame
-    return result;
+  public Frame newFrame(Frame prevFrame, SMethod method, Frame context) {
+    return this.universe.newFrame(prevFrame, method, context);
   }
 
-  private void popFrameAndPushResult(final SAbstractObject result) {
-    // Pop the top frame from the interpreter frame stack and compute the
-    // number of arguments
-    int numberOfArguments = popFrame().getMethod().getNumberOfArguments();
+  private final IndirectCallNode indirectCallNode = Truffle.getRuntime().createIndirectCallNode();
```

Comment:
Please order all fields at the top of the class (as per standard Java practice), I know, SOM does it at the bottom of the class, but I'd like to get away from that as well.